### PR TITLE
joyent/node-krill#8: toLDAPFilterString returns invalid LDAP search filter for empty predicates

### LINKED ISSUE #8: `toLDAPFilterString` returns invalid LDAP search filter for empty predicates
I feel embarrassed, as I should have caught that when writing tests, but the LDAP filter represented by `(*)` is invalid, and thus having the `toLDAPFilterString` method return such a filter when an empty predicate is passed is not desirable.

It seems that the most sensible thing to do would be to have `toLDAPFilterString` throw when such input is passed. We could provide an additional `isEmpty` method on the `Predicate` class's prototype that would return `true` if it's empty, and users could use that method to determine whether calling `toLDAPFilterString` is actually useful. This method would currently be identical to `trivial()`, but my understanding is that `trivial()` could evolve into something different in the future.

Anyway, making `toLDAPFilterString` throw on some input would be a backward incompatible change that would usually mandate a bump in the major version number. Since the current version of node-krill is in the 0.x range, and node-krill doesn't seem to have a significant numbers of dependents, I'm not sure what would be the most pragmatic solution.

What do you think about the proposal above?

### PR DESCRIPTION
**Pull request by misterdjules**
Fixes #8.

/cc @davepacheco @rmustacc

### CODE REVIEW

File: lib/krill.js [hunk 1]
```diff
@@ -624,8 +629,10 @@ function krillPrimPrintLDAP(pred) {
 	var key = sanityCheckResult.key;
 	var nbKeys = sanityCheckResult.nbKeys;
 
-	if (nbKeys === 0)
-		return ('(*)');
+	// Serializing an individual predicate with a number of keys !== 1 --
```
Comment (davepacheco) [inline lib/krill.js]: This module uses only C-style comments so far.  Would you mind using them here, too?

File: README.md [hunk 2]
```diff
@@ -83,6 +83,35 @@ console.log('LDAP search filter: ', predicate.toLDAPFilterString());
 /* Prints "(|(hostname=spike)(latency>300))" */
 ```
 
+Please note however that trivial predicates cannot be serialized to LDAP search
```
Comment (davepacheco) [inline README.md]: Does it make sense to explain that this is an LDAP limitation?  Something like: "Without knowing the LDAP object schema, it is not possible to generate a filter that matches all objects.  As a result, trivial predicates cannot be serialized as LDAP search filters."

**Comment by davepacheco** [comment]
Other than the two nits I mentioned, this looks good.

File: lib/krill.js (hunk 1 above)
Comment (misterdjules) [inline lib/krill.js]: I do not mind at all :)

File: README.md (hunk 2 above)
Comment (misterdjules) [inline README.md]: Your suggestion is much better, I will update that section.

**Comment by misterdjules** [comment]
@davepacheco Pushed a separate commit that I believe addresses the two comments above. Is it ok to merge?

**Comment by davepacheco** [comment]
Thanks.  Go for it!

**Comment by misterdjules** [comment]
Merged in a438574c2b0c9004fe98ba93449239e9776a372a. Thank you!

**Pull request closed without merging**